opam-ci-check/lint/opam_ci_check_lint.ml

```diff
@@ -271,13 +271,13 @@ module Checks = struct
         match (dune_constraint, dune_version) with
         | _, Error msg -> [ (pkg, DuneProjectParseError msg) ]
         | None, Ok None -> []
-        | Some "", _ -> [ (pkg, DuneLowerBoundMissing) ]
         | Some _, Ok None -> [ (pkg, DuneProjectMissing) ]
         | None, Ok (Some _) ->
           if is_dune (OpamPackage.name pkg) then []
           else [ (pkg, DuneDependencyMissing) ]
         | Some dep, Ok (Some ver) ->
-          if OpamVersionCompare.compare dep ver >= 0 then []
+          if dep = "" && ver = "1.0" then []
+          else if OpamVersionCompare.compare dep ver >= 0 then []
```

Comment:
This is a nit, so feel free to disregard --

While the logic here is pretty pretty straight forward, I have a nagging feeling it may not be as clear as we could get. Here's an idea to try to make the meaning here more apparent:

When `dep = ""`, we take this to mean that its dependency is a lower bound on `>= 1.0`. That is captured implicitly on line 279. We could make this explicit by adding a comment on this line, but I think we can make this explicit in the code, while also reducing the number of branches in the conditional, via

```suggestion
          let dune_dep_lower_bound = if dep = "" then "1.0" else dep in
          if OpamVersionCompare.compare dune_dep_lower_bound ver >= 0 then []
```